modules/yieldlabBidAdapter.js

```diff
@@ -517,14 +525,14 @@ function hasValidProperty(obj, propName) {
 }
 
 /**
- * Checks if an asset object is a main image.
- * A main image is defined as an image asset whose type value is 3.
+ * Checks if an asset object is of the given type.
  *
  * @param {Object} asset - The asset object to check.
- * @returns {boolean} Returns true if the object has a property img.type with a value of 3, otherwise false.
+ * @param {number} type - The type number to compare.
+ * @returns {boolean} Returns true if the object has a property img.type with a value of the given type, otherwise false.
  */
-function isMainImage(asset) {
-  return asset?.img?.type === 3
+function isImageAssetOfType(asset, type) {
```

Comment:
If we leave it generic and still want to shorten the calling place we could do something like:

```
function isImageAssetOfType(type) { // ICON = 1, MAIN = 3
  return asset = asset?.img?.type === type;
}
```

and then call it with `const iconAssetObj = find(assets, isImageAssetOfType(ICON));`

Otherwise just two methods as suggested above.